Frontend

IP1llar opened the pull request.

MarijaVitkauskaite committed:
feat: create new component and css folders with first files

MarijaVitkauskaite committed:
feat(app.css, app.tsx): change app.css and app.tsx files

MarijaVitkauskaite committed:
fix(main.tsx): change main.tsx to fix bug due to index.css being deleted

vadim-nest committed:
fix: rename .ts to .tsx. Start working on Learning Page

vadim-nest committed:
fix: change /ts to .tsx

MarijaVitkauskaite committed:
feat(app.tsx, welcome.tsx): create router navigation and add welcome to the app.tsx

vadim-nest committed:
style: add some fonts

MarijaVitkauskaite committed:
style(welcome.tsx, welcome.css, app.css): create top part of the welcome page

MarijaVitkauskaite committed:
feat: create welcome page with all the buttons

MarijaVitkauskaite committed:
feat: add navigation routes to the app.tsx

MarijaVitkauskaite committed:
feat: update components so app.tsx does not complain

MarijaVitkauskaite committed:
style: create navbar for the game page

MarijaVitkauskaite committed:
feat(navbar.tsx): add useLocation hook

IP1llar committed:
feat(utils/): implement graph datastructure and path-finding algorithms

IP1llar committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

MarijaVitkauskaite committed:
chore(index.html): change app title

MarijaVitkauskaite committed:
chore(index.html): update title

MarijaVitkauskaite committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

IP1llar committed:
feat(utils): implement graph datastructure and path-finding algorithms

vadim-nest committed:
style: add some svg to the learning page

vadim-nest committed:
style: some styling issues

MarijaVitkauskaite committed:
style: change p to h and add global fonts

IP1llar committed:
feat(client: utils and components): begin implementing maze component with maze logic and add tiles

vadim-nest committed:
style: create a basic learn page using svg

vadim-nest committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

MarijaVitkauskaite committed:
feat: change navbar to float on top of the pages and be fixed to the top

MarijaVitkauskaite committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

MarijaVitkauskaite committed:
fix: fix welcome page showing white line

IP1llar committed:
fix(app.css, maze.css, navbar.css, welcome.css): center navbar items and account for navbar height

majilaii committed:
feat(added a algo file in utils): sorting algos insertion bubble and selection

provided both each state of the array in each iteration plus the index that it has visited in trying
to sort the array

josepabellana committed:
feat(login button redirects to auth0): auth0 redirection implemented

josepabellana committed:
Co-authored-by: Kuong Ao Ieong <majilaii@users.noreply.github.com>

majilaii committed:
feat(algo sort ts): finished converting to ts and added some visited indexes to selecton sort

lol

josepabellana committed:
refactor(cleaning code login page): leaving Login page to mantaing /login

MarijaVitkauskaite committed:
feat: fix the welcome screen to the window, start game stats bar

MarijaVitkauskaite committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

MarijaVitkauskaite committed:
style: create game stats bar

MarijaVitkauskaite committed:
fix: fix navbar in the learning section

vadim-nest committed:
feat: create Sort Lesson page

IP1llar committed:
feat(maze): implement maze generation on client with slow animation

MarijaVitkauskaite committed:
fix: fix PLAY and LEARN to buttons

MarijaVitkauskaite committed:
fix: add cursor to buttons

josepabellana committed:
feat(redux first commit): install redux file

josepabellana committed:
feat(first redux commit): redux

MarijaVitkauskaite committed:
feat: check if user authenticated in, give access to PROFILE

majilaii committed:
feat(sort-lesson): implementated basic sorting of states in the front end

majilaii committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

IP1llar committed:
feat(maze.tsx, mazetile.tsx, mazetile.css): implement faster animation

Animation speeds up at the start and slows down at the end

MarijaVitkauskaite committed:
feat: dynamic navbar inc PROFILE

josepabellana committed:
feat(redux implementation working with counterstate mock): mock first variable on Redux

josepabellana committed:
feat(redux mocked example works): redux mock example works

MarijaVitkauskaite committed:
feat: colors as variables

MarijaVitkauskaite committed:
fix: white color change

MarijaVitkauskaite committed:
style: fix white color

IP1llar committed:
feat(game, maze, toolbar): implement a toolbar that has buttons to zoom the maze

IP1llar committed:
feat(game, maze, toolbar, minion): generate minion component, enable clicking of tile and minion

Game keeps track of which minion and which tile has been clicked most recently

vadim-nest committed:
style: add maze .svgs

vadim-nest committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

IP1llar committed:
style(maze): add border

MarijaVitkauskaite committed:
feat: make website responsive start

IP1llar committed:
style(app css): add box-sizing and background colour for body

josepabellana committed:
feat(redux and backend connection): backend connection

josepabellana committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

josepabellana committed:
feat(fixing bugs on navbar view): fixing navbar bugs

majilaii committed:
feat(sorting lesson): implemented bubble sort onto the divs

majilaii committed:
no cz

josepabellana committed:
refactor(adding backend query): profile request backend data

majilaii committed:
feat(his day is ruined): ruined isaacs day

majilaii committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

josepabellana committed:
feat(adding userdata type): redux reducer function added

MarijaVitkauskaite committed:
feat: make responsive

MarijaVitkauskaite committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

majilaii committed:
feat(insertion sort): insertion sort

josepabellana committed:
feat(auth user data show in profile): profile updated

MarijaVitkauskaite committed:
feat: responsive navbar done

majilaii committed:
feat(selection sort): selection sort implemented

josepabellana committed:
feat(add username edit button): username added in profile

majilaii committed:
refactor(sortinglessons file): changed file structure

josepabellana committed:
feat(add username id to maze): add username to maze

vadim-nest committed:
style: style minions a bit

IP1llar committed:
feat(maze, game): implement path finding for minions

josepabellana committed:
feat(add logo into userprofile): add avatar for users

josepabellana commented on Client/src/components/auth0Provider.tsx:
```diff
@@ -0,0 +1,23 @@
+
+  return (
+    <Auth0Provider
+      domain={"dev-mujh303ammb4fy01.uk.auth0.com"} //TODO add to .env file(auth0 joseph's domain)
```
Need to add this in a separate environment file

josepabellana commented on Client/src/components/game.tsx:
```diff
@@ -0,0 +1,127 @@
+    }
+  }, [currentMinion])
+
+  useEffect(() => {
```
this should be add in a separate file, maybe in a functionalities folder and made a function.

IP1llar replied:
👍

josepabellana commented on Client/src/components/game.tsx:
```diff
@@ -0,0 +1,127 @@
+  function addNewMinion() {
+    const newId = Object.keys(minions).length;
+    setMinions(prevMinions => {
+      return {
```
I recommend to create a file outside with the class Minion and redirect these calls there

IP1llar replied:
will do 👍

josepabellana commented on Client/src/components/login.tsx:
```diff
@@ -0,0 +1,17 @@
+function Login(props:any) {
+  
+  return(
+    <Auth0Provider
```
this file is only necessary IF someone access to the website with the /login, in my opinion. This can be deleted in the future,

MarijaVitkauskaite replied:
Done

josepabellana commented on Client/src/components/maze.tsx:
```diff
@@ -0,0 +1,93 @@
+
+function Maze({boxSize, setMazeCompleted, setCurrentMinion, minions, setCurrentTile, currentGraph, setCurrentGraph, height, width}: {
+  boxSize:number, 
+  height: number,
```
Task with creation of a Type as DOD. I'm thinking about having an external class Maze to externalize all this actions

IP1llar replied:
What does DOD stand for?

josepabellana replied:
Definition Of Done

josepabellana commented on Client/src/components/navbar.tsx:
```diff
@@ -0,0 +1,186 @@
+             {!user && (
+              <button
+                className="button"
+                onClick={() => {
```
the trigger on checking if the user is authenticated or loading can be done on a separate Component -NEW TASK

MarijaVitkauskaite replied:
Changed

josepabellana commented on Client/src/components/navbar.tsx:
```diff
@@ -0,0 +1,186 @@
+          <div className={`modal-body open-${modalOpen}`}>
+            {location.pathname === '/' && (
+              <button
+                className="button"
```
We can create a function with a switch that does the toogleNavbar and Model at the end. and goes to the location depending on the argument passed - NEW TASK

MarijaVitkauskaite replied:
Done

josepabellana commented on Client/src/components/navbar.tsx:
```diff
@@ -0,0 +1,186 @@
+      if (isAuthenticated === true) {
+        const { user, message } = await apiService.profile();
+        console.log(user)
+        if (user) dispatch(refreshData(user));
```
This user is undefined if the call to the backend goes wrong, so then it displays the error messgae

MarijaVitkauskaite replied:
Wrapped it in a try catch

josepabellana commented on Client/src/components/profile.tsx:
```diff
@@ -0,0 +1,67 @@
+  useEffect(()=>{
+    const base64String:any = Buffer.from(user.avatar.data.data).toString("base64");
+    setImg(base64String);
+  },[user])
```
check if its needed to have the dependency on user

MarijaVitkauskaite replied:
Changed

josepabellana commented on Client/src/components/profile.tsx:
```diff
@@ -0,0 +1,67 @@
+  const [img,setImg] = useState()
+
+  useEffect(()=>{
+    const base64String:any = Buffer.from(user.avatar.data.data).toString("base64");
```
This can be stored in the redux store as a base64 so no need to useEffect here. Change reducer action - NEW TASK

MarijaVitkauskaite replied:
Changed

josepabellana commented on Client/src/components/sortingLessons/bubble-lesson.tsx:
```diff
@@ -0,0 +1,97 @@
+
+  async function bubbleSort() {
+    setClicked(true)
+    const copyArr = array.slice();
```
[...array]

josepabellana commented on Client/src/components/sortingLessons/bubble-lesson.tsx:
```diff
@@ -0,0 +1,97 @@
+      
+  }
+  function delay(time:number) {
+    return new Promise((res) => setTimeout(res, time));
```
Thanks Josep for the idea.

No worries, Isaac told me this 1 month and a half ago

josepabellana replied:
ps. This must be added to a functionalities file to me reused with all the animations.

josepabellana commented on Client/src/components/sortingLessons/insertion-lesson.tsx:
```diff
@@ -0,0 +1,71 @@
+        secondP: "As we go through each element, if the current element is smaller than the previous one, the elements are swapped"
+      };
+
+    async function insertionSort() {
```
extracted in a functionalities folder

josepabellana commented on Client/src/components/sortingLessons/selection-lesson.tsx:
```diff
@@ -0,0 +1,73 @@
+        secondP: "As we go through each element, if the current element is smaller than the previous one, the elements are swapped"
+      };
+
+    async function selectionSort() {
```
extracted in a functionalities folder

josepabellana commented on Client/src/css/mazeTile.css:
```diff
@@ -0,0 +1,84 @@
+}
+
+.showNone {
+  background-image: none !important;
```
good use

josepabellana commented on Client/src/main.tsx:
```diff
@@ -1,10 +1,65 @@
+      {
+        path: '/login',
+        element: <Login />,
+      },
```
As I said, login can be deleted in a future iteration

josepabellana commented on Client/src/main.tsx:
```diff
@@ -1,10 +1,65 @@
+      },
+      {
+        path: '/bubbleLesson',
+        element: <BubbleLesson />,
```
this paths should be nested into the learning component

josepabellana replied:
/learning/whatever link

josepabellana commented on Client/src/services/apiService.ts:
```diff
@@ -0,0 +1,31 @@
+    method: 'GET',
+    credentials: 'include',
+    headers: { "Access-Control-Allow-Origin": '*', 'Content-Type': 'application/json' }
+  })
```
well done josep, remember to add a new query to retrieve avatars and keep the aPi service type in a types folder

majilaii replied:
yep keep up the good work josep!

josepabellana commented on Client/src/utils/graph.ts:
```diff
@@ -0,0 +1,92 @@
+
+export type value = string | number;
+
+export class Graph {
```
very well done

josepabellana commented on Client/src/utils/graph.ts:
```diff
@@ -0,0 +1,92 @@
+      path = dFS(valueX, valueY, this);
+      break;
+    }
+    if (path) return path;
```
very smooth, I like it a lot!

josepabellana commented on Client/src/utils/sorting-algo.ts:
```diff
@@ -0,0 +1,61 @@
+  let hasChanged = true;
+  while (hasChanged) {
+    hasChanged = false;
+    for (let i = 0; i <= arr.length - 2; i++) {
```
< arr.length -1

josepabellana replied:
this can be Improved if every time you go throw the loop substract one to length because in the first iteration you make sure the biggest will be at the end, at the second the second biggest... and soo...

josepabellana replied:
`function bubbleSort(arr) {
  let swaps = false;
  let iterations = 0;
  while (swaps === false) {
        swaps = true;
    for (let i = 0; i < arr.length-1-iterations; i++) {
      if(arr[i]>arr[i+1]){
        swaps = false;
        [arr[i],arr[i+1]] =[arr[i+1],arr[i]];
      } 
    }
    iterations++
  }
}`

josepabellana replied:
This code take into account iterations... Much faster

josepabellana commented on Client/src/utils/sorting-algo.ts:
```diff
@@ -0,0 +1,61 @@
+}
+
+export function insertionSortAlgo(arr: number[]) {
+  const ArrayStates: number[][] = [];
```
This implementation can be improved, this is n^2 complexity

josepabellana replied:
This implementation is n log n
`function insertionSort(arr) {
  for (let i = 1; i < arr.length; i++) {
    let j = i;
    while (arr[j] < arr[j - 1] && j > 0) {
      [arr[j - 1], arr[j]] = [arr[j], arr[j - 1]];
      j--;
    }
  }
  return arr;
}`

josepabellana reviewed (commented):
Haven't looked into the Maze logic but overall very good job. Watch comments for task or improvements

IP1llar commented on Client/index.html:
```diff
@@ -2,9 +2,9 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-    <title>Vite + React + TS</title>
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@300&family=Staatliches&display=swap" rel="stylesheet">
+    <title>aMAZEthing</title>
```
👍

IP1llar commented on Client/package.json:
```diff
@@ -9,14 +9,21 @@
   "devDependencies": {
     "@types/react": "^18.0.24",
     "@types/react-dom": "^18.0.8",
+    "@types/react-modal": "^3.13.1",
```
This should be uninstalled if it's not being used

IP1llar commented on Client/src/components/learning.tsx:
```diff
@@ -0,0 +1,141 @@
+      <div className='path-finding-algs'>
+        <h3>PATH FINDING ALGORITHMS</h3>
+        <div className='curve-1'>
+          <svg viewBox='-10 4 150 24'>
```
Can this svg be given its own component?

MarijaVitkauskaite replied:
Done

IP1llar commented on Client/src/components/navbar.tsx:
```diff
@@ -0,0 +1,186 @@
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [toggle, setToggle] = useState(false);
+  const [modalOpen, setModalOpen] = useState(false);
```
What's the difference between toggle and modalOpen? I think they're always toggled at the same time so could be combined

MarijaVitkauskaite replied:
Done

IP1llar commented on Client/src/components/welcome.tsx:
```diff
@@ -0,0 +1,16 @@
+  return (
+    <div className='welcome-page'>
+      <h1 className='welcome'>WELCOME.</h1>
+      <button className='options' onClick={() => navigate('/game')}>PLAY</button>
```
Can these be h1 instead of buttons?

MarijaVitkauskaite replied:
Accessibility, at least somewhere in our site

IP1llar commented on Client/src/features/user_slice.ts:
```diff
@@ -0,0 +1,54 @@
+import { createSlice, PayloadAction} from '@reduxjs/toolkit'
+interface UserState{ //shape of the state inside inside of the slice
```
Will this be needed elsewhere? If so we can export it

IP1llar commented on Client/src/utils/path-finding-algo.ts:
```diff
@@ -0,0 +1,270 @@
+import { Graph, value } from "./graph";
```
I've only tested bFS and vBFS so far

IP1llar commented on Client/tsconfig.json:
```diff
@@ -5,7 +5,7 @@
     "allowJs": false,
     "skipLibCheck": true,
-    "esModuleInterop": false,
+    "esModuleInterop": true,
```
This is to allow ts-node to run

IP1llar reviewed (commented):
Good effort all around

MarijaVitkauskaite committed:
feat: game stats bar made responsive for 3 sizes

majilaii committed:
feat(sorting algos): made it more visually concise

now you can see exactly which index the algorithm is going through before it swaps.

MarijaVitkauskaite committed:
feat: make welcome page responsive 3 sizes

MarijaVitkauskaite committed:
style: make 1.5 line height for readability

MarijaVitkauskaite committed:
style: update bubble sort lesson design

MarijaVitkauskaite committed:
style: add new style to bubble sort lesson

MarijaVitkauskaite committed:
style: style update and comment in bubble sort lesson

IP1llar committed:
feat(maze, minions, path): implement path visualisation

majilaii committed:
feat(sorting algos): made sorting algos accept repeated values now

vadim-nest committed:
style: add svg minions

josepabellana committed:
refactor(user profile bug fixed): add new function to redux reducer

majilaii committed:
feat(sorting algo): added random array generator and refresh page
